src/Description/ResourceEnhancer.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2017 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\SonataPhpcrAdminIntegrationBundle\Description;
+
+use Doctrine\Common\Util\ClassUtils;
+use Sonata\AdminBundle\Admin\Pool;
+use Symfony\Cmf\Component\Resource\Description\Description;
+use Symfony\Cmf\Component\Resource\Description\DescriptionEnhancerInterface;
+use Symfony\Cmf\Component\Resource\Description\Descriptor;
+use Symfony\Cmf\Component\Resource\Puli\Api\PuliResource;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * Add links and meta-info from Sonata Admin.
+ *
+ * @author Daniel Leech <daniel@dantleech.com>
+ *
+ * @internal
+ */
+class ResourceEnhancer implements DescriptionEnhancerInterface
```

Comment:
All description enhancers describe resources. I would call this `SonataEnhancer` or the like, this specific enhancer implementation is about adding things that Sonata knows about our document (links, names, etc.)